packages/core/src/step.ts

```diff
@@ -73,13 +73,9 @@ export function createUseStep(ctx: WorkflowOrchestratorContext) {
         if (event.eventType === 'step_started') {
           // Step has started - so remove from the invocations queue (only on the first "step_started" event)
           if (!hasSeenStepStarted) {
-            const invocationsQueueIndex = ctx.invocationsQueue.findIndex(
-              (invocation) =>
-                invocation.type === 'step' &&
-                invocation.correlationId === correlationId
-            );
-            if (invocationsQueueIndex !== -1) {
-              ctx.invocationsQueue.splice(invocationsQueueIndex, 1);
+            // O(1) lookup and delete using Map
+            if (ctx.invocationsQueue.has(correlationId)) {
+              ctx.invocationsQueue.delete(correlationId);
```

Comment:
Don't think checking `.has()` first here does anything useful:

```suggestion
            ctx.invocationsQueue.delete(correlationId);
```